src/app/clusters/bindings/bindings.cpp

```diff
@@ -209,13 +262,37 @@ CHIP_ERROR BindingTableAccess::WriteBindingTable(const ConcreteDataAttributePath
         {
             return CHIP_IM_GLOBAL_STATUS(ConstraintError);
         }
-        AddBindingEntry(target, path.mEndpointId);
+        if (BindingTable::GetInstance().Size() == EMBER_BINDING_TABLE_SIZE)
+        {
+            return CHIP_IM_GLOBAL_STATUS(ResourceExhausted);
+        }
+        ReturnErrorOnFailure(SaveWithNewEntry(target, path.mEndpointId));
+        AddBindingEntry(BindingTableEntryFromTargetStruct(target, path.mEndpointId));
         return CHIP_NO_ERROR;
     }
     return CHIP_IM_GLOBAL_STATUS(UnsupportedWrite);
 }
 } // namespace
 
+namespace chip {
+
+void AddBindingEntry(const EmberBindingTableEntry & entry)
+{
+    if (entry.type == EMBER_UNICAST_BINDING)
+    {
+        CHIP_ERROR err = BindingManager::GetInstance().UnicastBindingCreated(entry.fabricIndex, entry.nodeId);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogProgress(
```

Comment:
should this be Error instead? Otherwise add a comment on why getting an err is just Progress logged. Is this expected in some cases?

Reply:
It's expected if the bound peer is offline. We'll retry the connection on-demand so it's not treated as an error.